packages/server-functions-vite-plugin/src/compilers.ts

```diff
@@ -0,0 +1,336 @@
+            chainParts.unshift(current.property.name)
+          }
+
+          // Get the base callee
+          let base = current.object
+          while (!babel.types.isIdentifier(base)) {
+            if (babel.types.isCallExpression(base)) {
+              base = base.callee as babel.types.Expression
+            } else if (babel.types.isMemberExpression(base)) {
+              base = base.object
+            } else {
+              break
+            }
+          }
+          if (babel.types.isIdentifier(base)) {
+            chainParts.unshift(base.name)
+          }
+        } else if (babel.types.isIdentifier(current)) {
+          chainParts.unshift(current.name)
+        }
+
+        if (chainParts.length > 0) {
+          return chainParts.join('_')
+        }
+      }
+
+      // Rest of the existing checks...
+      if (babel.types.isFunctionDeclaration(currentPath.node)) {
+        return currentPath.node.id?.name
+      }
+
+      if (babel.types.isIdentifier(currentPath.node)) {
+        return currentPath.node.name
+      }
+
+      if (
+        isVariableDeclarator(currentPath.node) &&
+        isIdentifier(currentPath.node.id)
+      ) {
+        return currentPath.node.id.name
+      }
+
+      if (
+        babel.types.isClassMethod(currentPath.node) ||
+        babel.types.isObjectMethod(currentPath.node)
+      ) {
+        if (babel.types.isIdentifier(currentPath.node.key)) {
+          return currentPath.node.key.name
+        }
+        if (babel.types.isStringLiteral(currentPath.node.key)) {
+          return currentPath.node.key.value
+        }
+      }
+
+      return null
+    })()
+
+    if (name) {
+      nameParts.unshift(name)
+    }
+
+    currentPath = currentPath.parentPath
+  }
+
+  return nameParts.length > 0 ? nameParts.join('_') : 'anonymous'
+}
+
+function makeFileLocationUrlSafe(location: string): string {
+  return location
+    .replace(/[^a-zA-Z0-9-_]/g, '_') // Replace unsafe chars with underscore
+    .replace(/_{2,}/g, '_') // Collapse multiple underscores
+    .replace(/^_|_$/g, '') // Trim leading/trailing underscores
+}
+
+export function compileServerFnClient(
+  opts: ParseAstOptions & {
+    getRuntimeCode: (opts: {
+      serverFnPathsByFunctionId: Record<
+        string,
+        { nodePath: babel.NodePath; functionName: string; functionId: string }
+      >
+    }) => string
+    replacer: (opts: { filename: string; functionId: string }) => string
+  },
+) {
+  const ast = parseAst(opts)
+  const serverFnPathsByFunctionId: Record<
+    string,
+    {
+      nodePath: babel.NodePath
+      functionName: string
+      functionId: string
+    }
+  > = {}
+  const counts: Record<string, number> = {}
+
+  babel.traverse(ast, {
+    Program: {
+      enter(programPath) {
+        const recordServerFunction = (nodePath: babel.NodePath) => {
+          const fnName = findNearestVariableName(nodePath)
+
+          const baseLabel = makeFileLocationUrlSafe(
```

Comment:
what if you have `foo.tsx` and `foo.ts`? since you strip away the extension, the baseLabel could collide